cpp/ql/test/library-tests/constants/addresses/addresses.ql

```diff
@@ -0,0 +1,16 @@
+import cpp
+
+from Expr e, string msg, LocalVariable var, Function f
+where
+  e = var.getInitializer().getExpr().getFullyConverted() and
+  var.getFunction() = f and
+  (
+    e.isConstant() and
+    f.getName() = "nonConstantAddresses" and
+    msg = "misclassified as constant"
+    or
+    not e.isConstant() and
+    f.getName() = "constantAddresses" and
+    msg = "misclassified as NOT constant"
+  )
+select e, var.getName(), msg
```

Comment:
FWIW, I really like this style of test, where the expectations are embedded in the source code, instead of being separated out into the ".expected" file. It doesn't work for every possible test, but when it works, it's much easier to understand and modify the tests without accidentally accepting a bad change.